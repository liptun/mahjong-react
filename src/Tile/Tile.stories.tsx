import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import Tile from './Tile';

export default {
  component: Tile,
  args: { orientation: 'bottom', kind: 'bambooOne' },
  argTypes: {
    orientation: {
      options: ['left', 'right', 'bottom', 'top'],
      control: { type: 'radio' },
    },
    kind: {
      options: ['bambooOne', 'circleOne', 'characterOne'],
      control: { type: 'radio' },
    },
  },
} as ComponentMeta<typeof Tile>;

const Template: ComponentStory<typeof Tile> = (args) => <Tile {...args} />;

export const Primary = Template.bind({});
