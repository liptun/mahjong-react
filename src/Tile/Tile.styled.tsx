import { styled } from '@stitches/react';

const scale = 28;
const tile = {
  width: 2.4 * scale,
  height: 3.2 * scale,
  depth: 0.6 * scale,
  padding: 0.3 * scale,
};

export const Wrapper = styled('div', {
  width: tile.width,
  height: tile.height,
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  background: '#f4f2e7',
  margin: '.1em',
  position: 'relative',
  padding: tile.padding,
  '&::after, &::before': {
    content: '',
    position: 'absolute',
    display: 'block',
  },
  '&::after': {
    width: tile.depth,
    height: '100%',
    left: '100%',
    top: tile.depth / 2,
    background: 'linear-gradient(90deg, #e8e5d8 60%, #2f792f 60%)',
    transform: 'skewY(45deg)',
  },
  '&::before': {
    content: '',
    position: 'absolute',
    top: '100%',
    left: tile.depth / 2,
    display: 'block',
    width: '100%',
    height: tile.depth,
    background: 'linear-gradient(#d5d1c2 60%, #2f792f 60%)',
    transform: 'skewX(45deg)',
  },
  variants: {
    orientation: {
      bottom: {},
      left: {
        width: tile.height,
        height: tile.width,
        '& img': {
          transform: 'rotateZ(90deg)',
        },
      },
      right: {
        width: tile.height,
        height: tile.width,
        '& img': {
          transform: 'rotateZ(-90deg)',
        },
      },
      top: {
        '& img': {
          transform: 'rotateZ(180deg)',
        },
      },
    },
  },
});

export const Vector = styled('img', { width: tile.width });
