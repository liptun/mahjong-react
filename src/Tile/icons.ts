import bambooOne from './assets/bamboo-1.svg';
import bambooTwo from './assets/bamboo-2.svg';
import bambooThree from './assets/bamboo-3.svg';
import bambooFour from './assets/bamboo-4.svg';
import bambooFive from './assets/bamboo-5.svg';
import bambooSix from './assets/bamboo-6.svg';
import bambooSeven from './assets/bamboo-7.svg';
import bambooEight from './assets/bamboo-8.svg';
import bambooNine from './assets/bamboo-9.svg';
import characterOne from './assets/character-1.svg';
import characterTwo from './assets/character-2.svg';
import characterThree from './assets/character-3.svg';
import characterFour from './assets/character-4.svg';
import characterFive from './assets/character-5.svg';
import characterSix from './assets/character-6.svg';
import characterSeven from './assets/character-7.svg';
import characterEight from './assets/character-8.svg';
import characterNine from './assets/character-9.svg';
import circleOne from './assets/circle-1.svg';
import circleTwo from './assets/circle-2.svg';
import circleThree from './assets/circle-3.svg';
import circleFour from './assets/circle-4.svg';
import circleFive from './assets/circle-5.svg';
import circleSix from './assets/circle-6.svg';
import circleSeven from './assets/circle-7.svg';
import circleEight from './assets/circle-8.svg';
import circleNine from './assets/circle-9.svg';
import dragonGreen from './assets/dragon-green.svg';
import dragonRed from './assets/dragon-red.svg';
import dragonWhite from './assets/dragon-white.svg';
import windEast from './assets/wind-east.svg';
import windNorth from './assets/wind-north.svg';
import windSouth from './assets/wind-south.svg';
import windWest from './assets/wind-west.svg';

const Icons = {
  bambooOne,
  bambooTwo,
  bambooThree,
  bambooFour,
  bambooFive,
  bambooSix,
  bambooSeven,
  bambooEight,
  bambooNine,
  characterOne,
  characterTwo,
  characterThree,
  characterFour,
  characterFive,
  characterSix,
  characterSeven,
  characterEight,
  characterNine,
  circleOne,
  circleTwo,
  circleThree,
  circleFour,
  circleFive,
  circleSix,
  circleSeven,
  circleEight,
  circleNine,
  dragonGreen,
  dragonRed,
  dragonWhite,
  windEast,
  windNorth,
  windSouth,
  windWest,
};
export default Icons;

export type TIcons = typeof Icons;

