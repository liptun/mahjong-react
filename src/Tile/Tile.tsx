import * as S from './Tile.styled';
import React, { FC } from 'react';
import icons, { TIcons } from './icons';
import { TOrientation } from '../types';

interface Props {
  kind: keyof TIcons;
  orientation?: TOrientation;
}

const Tile: FC<Props> = ({ kind, orientation = 'bottom' }) => {
  return (
    <S.Wrapper orientation={orientation}>
      <S.Vector src={icons[kind]} />
    </S.Wrapper>
  );
};

export default Tile;
