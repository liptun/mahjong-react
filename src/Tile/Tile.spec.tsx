/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import Tile from './Tile';

test('Hello', () => {
  const rendered = render(<Tile kind="bambooOne" />);
  expect(rendered.baseElement).toMatchSnapshot();
});
