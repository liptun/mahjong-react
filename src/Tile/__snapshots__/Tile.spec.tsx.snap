// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello 1`] = `
<body>
  <div>
    <div
      class="c-eXFyNG"
    >
      <img
        class="c-ckMSqc"
        src="[object Object]"
      />
    </div>
  </div>
</body>
`;
